examples/dataset/object_detection_3d/object_detection_3d/upload_results.py

```diff
@@ -0,0 +1,388 @@
+                )
+                results[i][current_class] = dict(tp=tps, fp=fps, fn=fns)
+                raw_results[i][current_class] = dict(tp=all_tps, fp=all_fps, fn=all_fns)
+
+        for i, record in enumerate(df.itertuples()):
+            result = results[i]
+            raw_result = raw_results[i]
+            TP = [sum(tp) for tp in zip(result["Car"]["tp"], result["Cyclist"]["tp"], result["Pedestrian"]["tp"])]
+            FP = [sum(fp) for fp in zip(result["Car"]["fp"], result["Cyclist"]["fp"], result["Pedestrian"]["fp"])]
+            FN = [sum(fn) for fn in zip(result["Car"]["fn"], result["Cyclist"]["fn"], result["Pedestrian"]["fn"])]
+            FP_2D = [record.raw_inferences_2d[j] for j, fp in enumerate(FP) if fp]
+            FP_3D = [
+                UnmatchedScoredBoundingBox3D.from_bbox(record.raw_inferences_3d[j], np.max(overlaps[i][j]))
+                for j, fp in enumerate(FP)
+                if fp
+            ]
+            TP_2D = [record.raw_inferences_2d[j] for j, tp in enumerate(TP) if tp]
+            TP_3D = [record.raw_inferences_3d[j] for j, tp in enumerate(TP) if tp]
+            FN_2D = [record.image_bboxes[j] for j, fn in enumerate(FN) if fn]
+            FN_3D = [
+                UnmatchedLabeledBoundingBox3D.from_bbox(record.velodyne_bboxes[j], np.max(overlaps[i][:, j]))
+                for j, fn in enumerate(FN)
+                if fn
+            ]
+            matched_inference = [
+                record.raw_inferences_2d[j]
+                for j, tps in enumerate(
+                    zip(raw_result["Car"]["tp"], raw_result["Cyclist"]["tp"], raw_result["Pedestrian"]["tp"]),
+                )
+                if sum(tps)
+            ]
+            unmatched_inference = [
+                record.raw_inferences_2d[j]
+                for j, fps in enumerate(
+                    zip(raw_result["Car"]["fp"], raw_result["Cyclist"]["fp"], raw_result["Pedestrian"]["fp"]),
+                )
+                if sum(fps)
+            ]
+            unmatched_ground_truth = [
+                record.image_bboxes[j]
+                for j, fns in enumerate(
+                    zip(raw_result["Car"]["fn"], raw_result["Cyclist"]["fn"], raw_result["Pedestrian"]["fn"]),
+                )
+                if sum(fns)
+            ]
+
+            sample_metrics.append(
+                dict(
+                    image_id=record.image_id,
+                    matched_inference=matched_inference,
+                    unmatched_ground_truth=unmatched_ground_truth,
+                    unmatched_inference=unmatched_inference,
+                    nInferences=len(record.raw_inferences_3d),
+                    nValidObjects=sum(1 for ignore in ignored_gts_combined[i] if not ignore),
+                    nMatchedInferences=len(TP_2D),
+                    nMissedObjects=len(FN_2D),
+                    nMismatchedInferences=len(FP_2D),
+                    thresholds=f1_optimal_thresholds[difficulty],
```

Comment:
This does not display nicely with current UI support, as it is a dictionary. However, included for 2 reasons
1. for direct CSV import, we currently don't have capability to read eval_config from json.
2. for SDK upload, nested object does not expand in detail viewer, and only on "precise hover" over model/eval drop down.